.NET/Microsoft.Recognizers.Text.Number/Japanese/Parsers/JapaneseNumberParser.cs

```diff
@@ -0,0 +1,344 @@
+                        {
+                            intNumber = Config.ZeroToNineMapJpn[intNumberChar];
+                        }
+
+                        result.Value = intNumber * 10;
+                    }
+                }
+            }
+            else if (extResultJpn.Data.ToString().Contains("Num"))
+            {
+                var doubleText = Config.PercentageRegex.Match(resultText).Value;
+
+                if (doubleText.Contains("k") || doubleText.Contains("K") || doubleText.Contains("ｋ") ||
+                    doubleText.Contains("Ｋ"))
+                {
+                    power = 1000;
+                }
+
+                if (doubleText.Contains("M") || doubleText.Contains("Ｍ"))
+                {
+                    power = 1000000;
+                }
+
+                if (doubleText.Contains("G") || doubleText.Contains("Ｇ"))
+                {
+                    power = 1000000000;
+                }
+
+                if (doubleText.Contains("T") || doubleText.Contains("Ｔ"))
+                {
+                    power = 1000000000000;
+                }
+
+                result.Value = GetDigitValue(resultText, power);
+            }
+            else
+            {
+                var doubleText = Config.PercentageRegex.Match(resultText).Value;
+                doubleText = ReplaceUnit(doubleText);
+
+                var splitResult = Config.PointRegexJpn.Split(doubleText);
+                if (splitResult[0] == "")
+                {
+                    splitResult[0] = "零";
+                }
+
+                var doubleValue = GetIntValue(splitResult[0]);
+                if (splitResult.Length == 2)
+                {
+                    if (Config.NegativeNumberSignRegex.IsMatch(splitResult[0]))
+                    {
+                        doubleValue -= GetPointValue(splitResult[1]);
+                    }
+                    else
+                    {
+                        doubleValue += GetPointValue(splitResult[1]);
+                    }
+                }
+
+                result.Value = doubleValue;
+            }
+
+            result.ResolutionStr = result.Value + @"%";
+            return result;
+        }
+
+        protected ParseResult OrdParse(ExtractResult extResultJpn)
+        {
+            var result = new ParseResult
+            {
+                Start = extResultJpn.Start,
+                Length = extResultJpn.Length,
+                Text = extResultJpn.Text,
+                Type = extResultJpn.Type
+            };
+
+            var resultText = extResultJpn.Text;
+            resultText = resultText.Substring(1);
+
+            result.Value = Config.DigitNumRegex.IsMatch(resultText)
+                ? GetDigitValue(resultText, 1)
+                : GetIntValue(resultText);
+            result.ResolutionStr = result.Value.ToString();
+
+            return result;
+        }
+
+        private double GetDigitValue(string intStr, double power)
```

Comment:
Isn't this also the same as in Chinese?

Reply:
Yes, I'll move this function to public class.